src/XenCert/StorageHandler.py

```diff
@@ -2763,6 +2765,14 @@ def __del__(self):
         XenCertPrint("Reached StorageHandlerHBA destructor")
         StorageHandler.__del__(self)
 
+
+class StorageHandlerFCOE(StorageHandlerHBA):
+    def __init__(self, storage_conf):
+        XenCertPrint("Reached StorageHandlerHBA constructor")
+        StorageHandler.__init__(self, storage_conf)
+        self.sr_type = "lvmofcoe"
```

Comment:
After you rebased Mark's code, please pay attention to test ControlPathStressTests() again, because in  the super's method of StorageHandler.ControlPathStressTests() is called directly in BlockStorageHandler..ControlPathStressTests(), it may lead to your "lvmofcoe" does not work in my opinion.

Reply:
Thanks for the tip

Reply:
Sorry about that Tom. You will either need to make the GFS2 tests conditional or subclass the higher class which is probably undesirable as you'll need some stuff from the hba class I guess.

Reply:
Another thing, 'storage_type' is already existing in storage_conf, so seems self.sr_type is redundant.
I am not familiar with storage tech for what's difference between hba and fcoe upon xapi, if the difference is very small I will suggest to discard the new class StorageHandlerFCOE and just put thing into StorageHandlerHBA.

Reply:
The `storage_type` param is now in the form `hba` rather than `lvmohba` so would need to append the `lvmo` bit in front each time I want to use it in a xe call which is a little ugly. Apart from the sr type I think the handlers are mostly identical so I will try to merge them together.

Reply:
@MarkSymsCtx What additional conditions should be put on when to run the GFS2 tests? (Currently I have been testing on 7.1.1 and it is all working fine)